heuristic/local_search/reinsertion.py

```diff
@@ -0,0 +1,31 @@
+from numpy.random import RandomState
+
+from heuristic.classes import Solution, Problem
+from heuristic.repair_operators import greedy_insert
+
+
+def reinsertion(current: Solution, rnd_state: RandomState) -> Solution:
+    """
+    Attempts to find an improvement in the current solution by removing a
+    customer and reinserting this customer at the optimal location. If an
+    improvement is found the new solution is returned.
+    """
+    problem = Problem()
+
+    for customer in range(problem.num_customers):
+        destroyed = current.copy()
+
+        route = destroyed.find_route(customer)
+        route.remove_customer(customer)
+
+        destroyed.unassigned.append(customer)
+
+        destroyed.routes = [route for route in destroyed.routes
```

Comment:
This is already handled by greedy_insert - right?

Reply:
No, the destroy operators have the decorator function, which makes sure there are no empty routes. The repair operators will therefore never get a Solution instance with empty routes as input. We could add this to the repair operators but it seems to me something would've gone wrong elsewhere if this is neccesary.